trace-mini-agent/src/env_verifier.rs

```diff
@@ -140,9 +144,13 @@ impl EnvVerifier for ServerlessEnvVerifier {
                     .verify_gcp_environment_or_exit(verify_env_timeout)
                     .await;
             }
-            trace_utils::EnvironmentType::AzureSpringApp => {
-                trace_utils::MiniAgentMetadata::default()
-            }
+            trace_utils::EnvironmentType::AzureSpringApp => trace_utils::MiniAgentMetadata {
+                azure_spring_app_hostname: env::var("HOSTNAME").ok(),
+                azure_spring_app_name: env::var("ASCSVCRT_SPRING__APPLICATION__NAME").ok(),
+                gcp_project_id: trace_utils::MiniAgentMetadata::default().gcp_project_id,
+                gcp_region: trace_utils::MiniAgentMetadata::default().gcp_region,
```

Comment:
Is it correct for GCP be part of `AzureSpringApp`? Not sure if I'm missing context.

Reply:
It's correct for how this was originally designed but I agree it's confusing to be setting properties that are ultimately not used in some environments.

This `MiniAgentMetadata` struct previously had properties for GCP related fields and the mini agent version. I added the Azure Spring properties to it since that was the most convenient place to add this metadata. So we do need to initialize those GCP properties in the Azure Spring App and Azure Function environments even though they're not actually added to any span tags.

I'm planning on doing a refactor once Google Cloud Functions are supported with the sidecar agent and we can remove all of the GCP related logic in the mini agent.

https://github.com/DataDog/libdatadog/blob/8266d3801119ff45e0678b6cef6f8b9669adebe9/trace-utils/src/trace_utils.rs#L480-L486